specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

```diff
@@ -634,6 +634,39 @@
         "location"
       ]
     },
+    "BinaryDataset": {
+      "x-ms-discriminator-value": "Binary",
```

Comment:
Adding a new polymorphic child to an existing stable api version constitutes a breaking change.

Reply:
hey Amar, we've been adding new types all the time, and our customers are aware. If we had to create new API version for every type we would have tons API versions. It is not scalable.